File: Example/PDStrategy/PDStrategy/PDStrategy/TableView/PDTableViewHeaderFooter.m
//
//  PDTableViewHeaderFooter.m
//  PDStrategy
//
//  Created by Pavel Deminov on 19/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

#import "PDTableViewHeaderFooter.h"

@implementation PDTableViewHeaderFooter

- (void)setSectionInfo:(id<PDSectionInfo>)sectionInfo {
    _sectionInfo = sectionInfo;
    [self updateUI];
}

- (instancetype)initWithReuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithReuseIdentifier:reuseIdentifier];
    [self setup];
    return self;
}

- (void)updateUI {
    //override
}

- (void)setup {
    //override
}

- (void)setupForIB {
    //override
}

- (void)prepareForInterfaceBuilder {
    [self setup];
    [self setupForIB];
}

@end

File: Example/PDStrategy/PDStrategy/PDStrategy/TableView/PDTableViewHeaderFooter.swift
//
//  PDTableViewHeaderFooter.swift
//  PDStrategy
//
//  Created by Pavel Deminov on 04/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

import UIKit

class PDTableViewHeaderFooter: UITableViewHeaderFooterView, PDHeaderFooterInfo {

    var section: PDSectionInfo? {
        didSet {
            updateUI()
        }
    }
        
    override init(reuseIdentifier: String?) {
        super.init(reuseIdentifier: reuseIdentifier)
        setup()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        setup()
    }
    
    override func prepareForInterfaceBuilder() {
        setup()
        setupForIB()
    }
    
    func setup() {
        //override
    }
    
    func setupForIB() {
        //override
    }
    
    func updateUI() {
        //override
    }

}
